### PR TITLE
eval.c: clang/"Dead assignment"

### PR DESCRIPTION
**Pull request by bartop**

### CODE REVIEW

File: src/nvim/eval.c [hunk 1]
```diff
@@ -15827,11 +15827,11 @@ static void f_sign_getplaced(typval_T *argvars, typval_T *rettv, FunPtr fptr)
 {
   buf_T *buf = NULL;
   dict_T *dict;
-  dictitem_T *di;
```
Comment (bfredl) [inline src/nvim/eval.c]: `di` and `lnum` are the standard terms used in the codebase, for better and worse. I do not think we want to change them in isolation in a single function. (also for sign code in particular, we have yet some vim patches to merge, which we want to do first before doing refactors here).

**Comment by bartop** [comment]
Ok, no problem reverting these back.

niedz., 7 lip 2019, 23:56 użytkownik Björn Linse <notifications@github.com>
napisał:

File: src/nvim/eval.c (hunk 1 above)
Comment (bartop) [reply src/nvim/eval.c]: @bfredl Issue fixed as suggested

File: src/nvim/eval.c [hunk 2]
```diff
@@ -15850,11 +15850,10 @@ static void f_sign_getplaced(typval_T *argvars, typval_T *rettv, FunPtr fptr)
       }
       if ((di = tv_dict_find(dict, "lnum", -1)) != NULL) {
         // get signs placed at this line
-        lnum = (linenr_T)tv_get_number_chk(&di->di_tv, &notanum);
-        if (notanum) {
+        lnum = tv_get_lnum(&di->di_tv);
+        if (lnum <= 0) {
```
Comment (justinmk) [inline src/nvim/eval.c]: would be better to check -1 explicitly since that is the contract of `tv_get_number_chk`.

Comment (bartop) [reply src/nvim/eval.c]: Docstring says tv_get_lnum returns line number or 0 or 1.
https://github.com/neovim/neovim/blob/7d604a1e6158cda79f7ac485e0ef0e349014f8bf/src/nvim/eval/typval.c#L2761
Are you sure about that only -1 check is fine?

**Pull request merged**